src/connectors/dynamic-widgets/connectDynamicWidgets.ts

```diff
@@ -72,9 +78,8 @@ const connectDynamicWidgets: DynamicWidgetsConnector =
         init(initOptions) {
           widgets.forEach((widget) => {
             const attribute = getWidgetAttribute(widget, initOptions);
-            localWidgets.set(attribute, { widget, isMounted: true });
+            localWidgets.set(attribute, { widget, isMounted: false });
           });
-          initOptions.parent!.addWidgets(widgets);
```

Comment:
What is this replaced with?

Reply:
I was experimenting whether it was actually needed, either something was wrong with my local environment that kept the refinement, but they no longer seem to be writing an empty url initially without this line. I need to check more

Reply:
We found in #4849 that the write which happens on initial mount (before the widgets are added) was caused by the insights middleware, and not normal behaviour